dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/entities/MessageEntity.kt

```diff
@@ -0,0 +1,65 @@
+package org.dataland.datalandcommunitymanager.entities
+
+import jakarta.persistence.Column
+import jakarta.persistence.Entity
+import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.ManyToOne
+import jakarta.persistence.Table
+import org.dataland.datalandcommunitymanager.model.dataRequest.StoredDataRequestMessageObject
+import org.dataland.datalandemail.email.isEmailAddress
+import java.time.Instant
+import java.util.UUID
+
+/**
+ * The database entity for storing a single message in a data request conversation
+ */
+@Entity
+@Table(name = "messages")
+data class MessageEntity(
+    @Id
+    val messageId: String,
+
+    @Column(columnDefinition = "TEXT")
+    val contacts: String,
+
+    @Column(columnDefinition = "TEXT")
+    val message: String?,
+
+    val creationTimestamp: Long,
+
+    @ManyToOne(optional = false)
+    @JoinColumn(name = "data_request_id")
+    var dataRequest: DataRequestEntity,
+) {
+    companion object {
+        private const val emailSeparator = ";"
+    }
+
+    init {
+        require(contacts.isNotEmpty())
+        require(contacts.split(emailSeparator).all { it.isEmailAddress() })
+        require(message?.isNotBlank() ?: true)
+    }
+
+    constructor(
+        messageObject: StoredDataRequestMessageObject,
+        dataRequest: DataRequestEntity,
+    ) : this(
+        messageId = UUID.randomUUID().toString(),
+        contacts = messageObject.contacts.joinToString(emailSeparator),
+        message = messageObject.message,
+        creationTimestamp = messageObject.creationTimestamp ?: Instant.now().toEpochMilli(),
```

Comment:
Ist der part mit dem Fragezeichen sicher notwendig?

Ich glaube dort, wo wir StoredDataReuqestMEssageObjects bauen, passen wir eigentlich immer einen korrekten timestamp mit rein (und niemals null) oder?

Dann könnte "creationTimeStamp" auch non-nullable sein in "StoredDtaRequestMessageObject".

Reply:
Yes. Nice. Done.